### PR TITLE
Merge files from master branch.

### PR DESCRIPTION
**Pull request by junaruga**
This PR is to merge below files on `master` branch to `devel` branch
* .gitignore
* .travis.yml

Related to https://github.com/trinityrnaseq/trinityrnaseq/issues/620 .

### CODE REVIEW

**Comment by junaruga** [comment]
.travis.yml

```
  - make || (show_log; false)
```

works like this when `make` fails with an error.

https://travis-ci.org/trinityrnaseq/trinityrnaseq/jobs/467761026#L1921-L1924

**Comment by junaruga** [comment]
Just in case, review please.

The Ci is passed on `devel` branch.
Total 4 commits. 1st commit is to merge from `master` to `devel` branch.
2nd to 4th commit is to fix issues for `devel` branch.

**Comment by junaruga** [comment]
Brian, do you like to review this PR?

**Comment by brianjohnhaas** [comment]
Will do.  Thx!  I'll aim to do this soon

-via googleFi

File: trinity-plugins/Makefile [hunk 1]
```diff
@@ -1,8 +1,8 @@
 
 OS := $(shell uname)
 
-CXX=g++
-CC=gcc
+CXX ?= g++
```
Comment (brianjohnhaas) [inline trinity-plugins/Makefile]: I vaguely remember this causing trouble for some installations, but don't remember all the specifics.  I had to revert this change from an earlier modification.  I suppose I could push it through again now, and then we can revisit it when it crops up again.

**Comment by brianjohnhaas** [comment]
apologies for the delay in pushing this through.

File: trinity-plugins/Makefile (hunk 1 above)
Comment (junaruga) [reply trinity-plugins/Makefile]: I see. Let's check the issue again when it will happen again in the future.

**Comment by junaruga** [comment]
Thanks for the merging! No problem!
